src/inverse-kinematics/src/InverseKinematics.cpp

```diff
@@ -1398,8 +1396,7 @@ namespace iDynTree {
 #endif
     }
 
-    // this should be const reference but we keep it like this to avoid breaking the API. Please change me in iDynTree 4.0
-    void InverseKinematics::setCOMConstraintProjectionDirection(iDynTree::Vector3 direction)
+    void InverseKinematics::setCOMConstraintProjectionDirection(const iDynTree::Vector3 direction)
```

Comment:
```suggestion
    void InverseKinematics::setCOMConstraintProjectionDirection(const iDynTree::Vector3& direction)
```

Reply:
Not that I am against it, but consider that for such as small class, I am not sure if the advantage of passing by reference in term of avoiding copy  (pro) is more then the disavantage of having an additional indirection.